bciers/apps/compliance/src/app/components/compliance-summary/manage-obligation/apply-compliance-units/ApplyComplianceUnitsComponent.tsx

```diff
@@ -111,28 +139,90 @@ export default function ApplyComplianceUnitsComponent({
     }
   };
 
+  // First submission: Get compliance data and display it
   const handleSubmit = async (
     e: IChangeEvent<ApplyComplianceUnitsFormData>,
   ) => {
-    setIsSubmitting(true);
+    setStatus("submitting");
     const response = await actionHandler(
       `compliance/bccr/accounts/${e.formData?.bccr_holding_account_id}/compliance-report-versions/${complianceSummaryId}/compliance-units`,
+      "GET",
+      "",
+    );
+    if (!response || response.error) {
+      setStatus("idle");
+      setErrors([response.error || "Failed to get compliance units data"]);
+    } else {
+      // Set the outstanding balance from the response
+      setInitialOutstandingBalance(response.outstanding_balance || 0);
+
+      // Update form data with the full compliance data from the response
+      setFormData((prev: ApplyComplianceUnitsFormData) => {
+        // Create a clean data object for the compliance phase
+        const cleanFormData = {
+          bccr_holding_account_id: prev.bccr_holding_account_id,
+          bccr_trading_name: prev.bccr_trading_name,
+          ...response,
+        };
+
+        setCurrentPhase("compliance_data");
+        return cleanFormData;
+      });
+
+      setStatus("submitted");
+      setErrors(undefined);
+    }
+  };
+
+  // Second submission: Apply the compliance units
+  const handleApply = async () => {
+    setStatus("applying");
+    const response = await actionHandler(
+      `compliance/bccr/accounts/${(formData as ApplyComplianceUnitsFormData)
+        ?.bccr_holding_account_id}/compliance-report-versions/${complianceSummaryId}/compliance-units`,
       "POST",
       "",
       {
-        body: JSON.stringify(e.formData),
+        body: JSON.stringify(formData),
       },
     );
-    setIsSubmitting(false);
     if (!response || response.error) {
+      setStatus("submitted");
       setErrors([response.error || "Failed to apply compliance units"]);
     } else {
-      setIsSubmitted(true);
+      setStatus("applied");
       setErrors(undefined);
     }
   };
 
-  // Check if the selected units exceed or equal 50% of the initial outstanding balance
+  // Check if we should show the Submit button (when trading name is received)
+  const shouldShowSubmitButton = useMemo(() => {
+    const data = formData as ApplyComplianceUnitsFormData;
+    return !!(
+      data?.bccr_trading_name &&
+      (status === "idle" || status === "submitting")
+    );
+  }, [formData, status]);
+
+  // Check if Submit button should be enabled (when checkbox is checked)
+  const canSubmit = useMemo(() => {
+    const data = formData as ApplyComplianceUnitsFormData;
+    return !!(shouldShowSubmitButton && data?.confirmation_checkbox);
+  }, [formData, shouldShowSubmitButton]);
+
+  // Check if we should show the Apply button (after data is loaded)
+  const shouldShowApplyButton = useMemo(() => {
+    const data = formData as ApplyComplianceUnitsFormData;
+    const result = !!(
+      status === "submitted" &&
```

Comment:
We should account for both `submitted` and `applying` states to show the loading spinner while waiting for BCCR to transfer the units. Otherwise, the Apply button disappears without any feedback, which could be confusing for the user.